staging/src/k8s.io/apiserver/plugin/pkg/audit/webhook/webhook.go

```diff
@@ -229,15 +291,43 @@ L:
 		}
 	}
 
+	return events
+}
+
+// collectLastEvents assumes that the buffer was closed. It collects the first
+// maxBatchSize events from the closed buffer into a batch and returns them.
+func (b *batchBackend) collectLastEvents() []auditinternal.Event {
```

Comment:
> I think any opportunity we have to remove state the better!

I don't agree with that. Any private method can be made stateless. Stateless itself is not a goal, it should serve some purpose. In this particular case I think the private method with a meaningful name is better for readability than a stateless function

Let's let @sttts decide on that, as a tiebreaker :)